util.go

```diff
@@ -154,11 +155,43 @@ func trimeSchemaRefLinkPrefix(ref string) string {
 	return strings.TrimPrefix(ref, "#/components/schemas/")
 }
 
-func genSchemeaObjectID(pkgName, typeName string) string {
-	typeNameParts := strings.Split(typeName, ".")
-	pkgName = replaceBackslash(pkgName)
+func genSchemaObjectID(pkgName, typeName string, aliases map[string]*string) string {
```

Comment:
We could use a couple of unit tests for this function since it's doing quite a lot more now than it was previously. Should be a bit easier to do now that you're only passing in the alias dictionary, no need to build a mock parser struct👌